internal/characters/neuvillette/cons.go

```diff
@@ -0,0 +1,122 @@
+package neuvillette
+
+import (
+	"strings"
+
+	"github.com/genshinsim/gcsim/internal/common"
+	"github.com/genshinsim/gcsim/pkg/core/attacks"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/geometry"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+func (c *char) c1() {
+	if c.Base.Ascension < 1 {
+		return
+	}
+
+	c1 := NeuvA1Keys{event.OnCharacterSwap, "neuvillette-a1-c1-onfield"}
+	c.a1Statuses = append(c.a1Statuses, c1)
+	c.Core.Events.Subscribe(c1.Evt, func(args ...interface{}) bool {
+		next := args[1].(int)
+		if next == c.Index {
+			c.AddStatus(c1.Key, 30*60, false)
+		}
+		return false
+	}, c1.Key)
+}
+
+func (c *char) c2() {
+	if c.Base.Ascension < 1 {
+		return
+	}
+	c.AddAttackMod(character.AttackMod{
+		Base: modifier.NewBase("neuvillette-c2", -1),
+		Amount: func(atk *combat.AttackEvent, t combat.Target) ([]float64, bool) {
+			if strings.Contains(atk.Info.Abil, chargeJudgementName) {
+				c2Buff := make([]float64, attributes.EndStatType)
+				c2Buff[attributes.CD] = 0.14 * float64(c.countA1())
+				return c2Buff, true
+			}
+			return nil, false
+		},
+	})
+}
+
+func (c *char) c4() {
+	c.Core.Events.Subscribe(event.OnHeal, func(args ...interface{}) bool {
+		target := args[1].(int)
+
+		// 4s CD
+		if c.Core.F < c.lastc4+4*60 {
+			return false
+		}
+		if c.Core.Player.Active() != c.Index {
+			return false
+		}
+		if c.Index != target {
+			return false
+		}
+		if c.Core.Player.Active() == c.Index && c.Index == target {
+			// TODO: find the actual sourcewater droplet spawn shape for Neuv C4
+			player := c.Core.Combat.Player()
+			center := player.Pos().Add(player.Direction().Normalize().Mul(geometry.Point{X: 3.0, Y: 3.0}))
+			pos := geometry.CalcRandomPointFromCenter(center, 0, 2.5, c.Core.Rand)
+			common.NewSourcewaterDroplet(c.Core, pos, combat.GadgetTypSourcewaterDropletNeuv)
+			c.lastc4 = c.Core.F
```

Comment:
Similarly to C6, this could also use a character event log indicating that this proc'd.

Reply:
Please prepend "C4: " just like you did for the C6 log.